src/index.js

```diff
@@ -113,8 +146,8 @@ let Rosnodejs = {
         return Promise.resolve(this.getNodeHandle());
       }
       // else
-      throw new Error('Unable to initialize node [' + nodeName + '] - node ['
-                      + rosNode.getNodeName() + '] already exists');
+      return Promise.reject( Error('Unable to initialize node [' + nodeName + '] - node ['
```

Comment:
My main concern with this change is that it changes existing behavior.

Reply:
Since this was already returning a promise and the user was expecting that the promise would either resolve or reject (and set up the appropriate `.catch()`), I'm of the opinion that this should just reject instead of throwing an error, unless you want this to be explicitly fatal.

Reply:
That was my original thought - in my mind this was more explicit user-error. The distinction seems less noticeable in newer versions of JS though.